src/main/webapp/placeGuides/ListPlaceGuideDisplayer.js

```diff
@@ -2,75 +2,132 @@
+    this._listPlaceGuideDisplayerDiv = document.getElementById("listPlaceGuideDisplayer");
+    this._listPlaceGuideDisplayerDiv.classList.add(
+        "list-group", "my-place-guide-list", "list-place-guide-displayer", "form-card");
+    this._placeGuideDisplayQuery = placeGuideDisplayQuery;
+    this._listPlaceGuideDisplayerDiv.appendChild(
+        this.createListTitle(
+            ListPlaceGuideDisplayer.QUERY[placeGuideDisplayQuery]));
   }
 
-  addToList(placeGuideId) {
+  get listPlaceGuideDisplayerDiv() {
+    return this._listPlaceGuideDisplayerDiv;
+  }
 
+  update(placeGuides) {
+    this.removeAllPlaceGuidesFromList();
+    this._listPlaceGuideDisplayerDiv.appendChild(
+        this.createListTitle(
+            ListPlaceGuideDisplayer.QUERY[this._placeGuideDisplayQuery]));
+    this.addPlaceGuidesToList(placeGuides);
   }
 
-  removePreviousPlaceGuidesFromList(placeGuides) {
-    for (const placeGuideId in this._placeGuidesOnList) {
-      if (this._placeGuidesOnList.hasOwnProperty(placeGuideId)) {
-        if (!placeGuides.hasOwnProperty(placeGuideId)) {
-          // This placeGuide is not needed anymore.
-          this.remove(placeGuideId);
-        }
-      }
-    }
+  removeAllPlaceGuidesFromList() {
+    while (this._listPlaceGuideDisplayerDiv.firstChild) {
+      this._listPlaceGuideDisplayerDiv.removeChild(
+          this._listPlaceGuideDisplayerDiv.lastChild);
+    };
   }
 
-  addNewPlaceGuidesToList(placeGuides) {
-    for (const placeGuideId in placeGuides) {
+  addPlaceGuidesToList(placeGuides) {
+    for (var placeGuideId in placeGuides) {
       if (placeGuides.hasOwnProperty(placeGuideId)) {
-        if (!this._placeGuidesOnList.hasOwnProperty(placeGuideId)) {
-          // new placeGuide should be constructed.
-          this._placeGuidesOnList[placeGuideId] =
-              this.
-                  constructPlaceGuideOnListFromPlaceGuide(
-                      placeGuides[placeGuideId]);
-          this.addToList(placeGuideId);
-        }
+        var constructedPlaceGuideOnListDiv = 
+            this.constructPlaceGuideOnListDivFromPlaceGuide(placeGuides[placeGuideId]);
+        this._listPlaceGuideDisplayerDiv.appendChild(constructedPlaceGuideOnListDiv);
       }
     }
   }
 
   remove(placeGuideId) {
-    // Delete from list.
-    this.removeFromList(placeGuideIde);
-    // Delete from memory.
-    delete this._placeGuidesOnList[placeGuideId];
+    const placeGuideDivId = "placeGuideOnList-" + "{" + placeGuideId + "}";
+    const placeGuideDiv = document.getElementById(placeGuideDivId);
+    if (this._listPlaceGuideDisplayerDiv.contains(placeGuideDiv)) {
+      this._listPlaceGuideDisplayerDiv.removeChild(placeGuideDiv);
+    }
+    return placeGuideDiv;
   }
 
+   // Move the highlighted place guide to top of list.
   highlight(placeGuideId) {
-
+    const placeGuideDiv = this.remove(placeGuideId);
+    this._listPlaceGuideDisplayerDiv.removeChild(document.getElementById("listTitle"));
```

Comment:
Again, don't remove the title. Find the next sibling and insert before that one for example. Or create a separate parent div for the placeguides which does not contain the title.

Reply:
Modified the function so that it doesn't remove the title when highlighting, instead, it will just append the highlighted div after the title.